### PR TITLE
npm run lint command to lint HTML, CSS, and JS

### LINKED ISSUE #14: Sample app needs npm run lint command to lint HTML, CSS, and JS

### LINKED ISSUE #66: Swap jshint for eslint in sample app

### PR DESCRIPTION
**Pull request by lyomr**
This PR

- Adds lint command for HTML, CSS, JS in SampleApp
- Replaces jshint for eslint in SampleApp
- adds husky instead of precommit
- closes #14 and closes #66

### CODE REVIEW

File: generators/app/templates/templatedPackage.json [hunk 1]
```diff
@@ -18,7 +18,9 @@
     "roosevelt-closure": "^0.3.0"
   },
   "devDependencies": {
-    "jshint": "*",
+    "eslint": "*",
+    "lesshint": "*",
+    "htmlhint": "*",
```
Comment (kethinov) [inline generators/app/templates/templatedPackage.json]: version these using `^`

File: generators/app/templates/templatedPackage.json [hunk 2]
```diff
@@ -88,6 +114,10 @@
     "start": "npm run prod",
     "prod": "./node_modules/.bin/supervisor -k -e js,json,less,html -i statics/.build/,.git/,public/,node_modules/ -- app.js -prod",
     "dev": "./node_modules/.bin/supervisor -k -e js,json,less,html -i statics/.build/,.git/,public/,node_modules/ -- app.js -dev",
-    "test": "./node_modules/.bin/jshint app.js"
+    "test": "npm run lint app.js",
```
Comment (kethinov) [inline generators/app/templates/templatedPackage.json]: remove app.js from this command

File: generators/app/templates/templatedPackage.json [hunk 3]
```diff
@@ -88,6 +113,9 @@
     "start": "npm run prod",
     "prod": "./node_modules/.bin/supervisor -k -e js,json,less,html -i statics/.build/,.git/,public/,node_modules/ -- app.js -prod",
     "dev": "./node_modules/.bin/supervisor -k -e js,json,less,html -i statics/.build/,.git/,public/,node_modules/ -- app.js -dev",
-    "test": "./node_modules/.bin/jshint app.js"
```
Comment (kethinov) [inline generators/app/templates/templatedPackage.json]: don't delete the whole line

just remove app.js from the command

Comment (kethinov) [inline generators/app/templates/templatedPackage.json]: the line should read `"test": "npm run lint`

**Comment by alallier** [comment]
Both the stylelint and htmlhint commands are looking for config files, those will need to be included in order for them to work properly

File: generators/app/templates/templatedPackage.json [hunk 4]
```diff
@@ -88,6 +115,15 @@
...
-  }
+    "test": "npm run lint",
+    "eslint": "./node_modules/.bin/eslint .",
+    "stylelint": "./node_modules/.bin/stylelint \"statics/css/**/*.less\" --syntax less",
+    "htmlhint": "./node_modules/.bin/htmlhint --config .htmlhintrc mvc/views",
+    "lint": "npm run htmlhint && npm run stylelint && npm run eslint"
+  },
+  "pre-commit": [
```
Comment (kethinov) [inline generators/app/templates/templatedPackage.json]: for this to work we need to include pre-commit in the package.json

**Comment by lyomr** [comment]
This PR is part of #69

**Pull request closed without merging**